### DISCUSSION TITLE
Migrations locally

### DISCUSSION THREAD

**Discussion by barbinbrad** [supabase/supabase#11591] [Questions]
Apologies for the dumb question, I'm sure this has been answered already, but I haven't been able to track it down. 

Is there a way to run a migration locally through the supabase CLI? Currently, I'm using prisma for migrations and supabase CLI to generate types and start the containers locally. But when I run `supabase db push` it asks for a project ref. I'm looking for something like `prisma migrate dev`. Does this exist, or should I stick with prisma for manual migrations?

**Comment by silentworks** [comment]
Are you asking how to run the migrations you have in your supabase CLI local project to the remote database? If yes you first need to link your local supabase project to the remote (hosted) project.

> I'm assuming you have a local supabase setup using the CLI already e.g. `supabase init`

```sh
supabase login
supabase link --project-ref [proj-ref]
```
Visit https://app.supabase.com/project/_/settings/general and you will find your **Reference ID** there, replace `[proj-ref]` with the **Reference ID**. Next you can create a migration using the supabase cli

```sh
supabase migration new [migration_name]
```
This will create a new migration file inside of `supabase/migrations`, you can now manually write the migration using SQL. The other option is to use the dashboard locally to create your table along with its columns and then use the CLI to generate a migration file based on that.

```sh
supabase db diff --use-migra -f [migration_name]
```

You can read more about these in our guides and in the reference docs.

https://supabase.com/docs/guides/resources/supabase-cli/local-development
https://supabase.com/docs/guides/resources/supabase-cli/managing-environments
https://supabase.com/docs/reference/cli/supabase-db-diff
https://supabase.com/docs/reference/cli/supabase-migration

**Reply by barbinbrad** [reply]
Hey, thanks for taking the time to write! Really appreciate it. I'm asking how to run a migration to my local database.

**Reply by barbinbrad** [reply]
I realize you're probably thinking -- the whole point of supabase studio is that you don't have to do this. But the way I was thinking about it is that I want to make sure my migration scripts work on my local database before pushing remote.

**Reply by barbinbrad** [reply]
I see in one of your links that `supabase db reset` is what I'm looking for, thanks again.
